From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Hridya Valsaraju <hridya@google.com>
Date: Mon, 21 Dec 2020 11:09:12 -0800
Subject: ANDROID: defconfig: Enable DMA-BUF system heap as a module

Unset CONFIG_DMABUF_HEAPS_SYSTEM from gki_defconfig
so that it can be enabled as a vendor module.

This change is intended to allow partners to continue using
device/SoC-specific optimizations in their system heap implementations
when they switch from ION to DMA-BUF heaps. The ION system heap was
built-in and partners were asked to override the system heap ID with
their own if they wanted to override the GKI system heap. This
per-vendor approach to override system heap will no longer be possible
with DMA-BUF heaps since Android S aims to restrict framework access to
DMA-BUF vendor heaps by only letting it access ABI-defined vendor heaps.

VTS tests will be created to ensure that the system heap is present
at /dev/dma_heap/system and behaves as expected of the system heap.

Bug: 175697666
Bug: 155218010
Bug: 172527615

Signed-off-by: Hridya Valsaraju <hridya@google.com>
Change-Id: Id97ed4814517339c69b99f8135e9a66e59d951a9
(cherry picked from commit 628ed925820bd14b93d30133830f8af2018ac6ba)
Signed-off-by: Palmer Dabbelt <palmerdabbelt@google.com>
---
 arch/arm64/configs/db845c_gki.fragment | 1 +
 arch/arm64/configs/gki_defconfig       | 1 -
 arch/x86/configs/gki_defconfig         | 1 -
 3 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm64/configs/db845c_gki.fragment b/arch/arm64/configs/db845c_gki.fragment
--- a/arch/arm64/configs/db845c_gki.fragment
+++ b/arch/arm64/configs/db845c_gki.fragment
@@ -119,6 +119,7 @@ CONFIG_SOUNDWIRE=m
 CONFIG_SOUNDWIRE_QCOM=m
 CONFIG_SLIMBUS=m
 CONFIG_SLIM_QCOM_NGD_CTRL=m
+CONFIG_DMABUF_HEAPS_SYSTEM=m
 # CONFIG_CXD2880_SPI_DRV is not set
 # CONFIG_MEDIA_TUNER_SIMPLE is not set
 # CONFIG_MEDIA_TUNER_TDA18250 is not set
diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -451,7 +451,6 @@ CONFIG_RTC_CLASS=y
 CONFIG_RTC_DRV_PL030=y
 CONFIG_RTC_DRV_PL031=y
 CONFIG_DMABUF_HEAPS=y
-CONFIG_DMABUF_HEAPS_SYSTEM=y
 CONFIG_UIO=y
 CONFIG_VHOST_VSOCK=y
 CONFIG_STAGING=y
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -406,7 +406,6 @@ CONFIG_LEDS_TRIGGER_TRANSIENT=y
 CONFIG_EDAC=y
 CONFIG_RTC_CLASS=y
 CONFIG_DMABUF_HEAPS=y
-CONFIG_DMABUF_HEAPS_SYSTEM=y
 CONFIG_UIO=y
 CONFIG_VHOST_VSOCK=y
 CONFIG_STAGING=y
